### ISSUE TITLE
code autocomplete inserts extra characters

### ISSUE THREAD

**Issue by mmmatjaz** [spyder-ide/spyder#11237]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ ] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder
    * [x] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [x] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
code autocomplete inserts additional characters that look like markup:
"def __init__(${1:self})$0" instead of "def __init__(self)"
tested on two machines with miniconda (windows and ubuntu)

### What steps reproduce the problem?

1.  type "def __i" under class definition 
2.  press ctrl+space
3.  "def __init__(self)" is correctly suggested
4.  when suggestion is selected, "def __init__(${1:self})$0" is inserted instead of "def __init__(self)"

### What is the expected output? What do you see instead?
when suggestion is selected, "def __init__(${1:self})$0" is inserted instead of "def __init__(self)"

### Paste Traceback/Error Below (if applicable)

```python-traceback

PASTE TRACEBACK HERE

```

## Versions

* Spyder version: 4.0.0
* Python version: 3.6
* Qt version: 
* PyQt version: 
* Operating System name/version: 

### Dependencies

```
PASTE DEPENDENCIES HERE
```

**Comment by plungg** [comment]
The Kite team saw a piece of feedback on this issue as well.
"If you have typed more than the minimum number of characters required for Kite to suggest a completion, Kite will insert the suggestion before the characters typed after the minimum count. This results in errors most of the time and using completions slows down my typing speed rather than speeding it up and simplifying it."

**Comment by leavor** [comment]
I encountered the same problem, when the function starting with an underscore and this function had parameters, there was a problem with function code completion.

See #10924

**Comment by ricardobarriosp** [comment]
I also can report having the same problem in Spyder 4.0.1. In the code below. 

I tried the  simple code below. Instead of getting the proper function signature after accepting the suggestion, I also got some unwanted text.
```
class Test():
    def __init__(${1:self})$0:        
    def __format__(${1:self}, format_spec):
    def __sizeof__(${1:self})$0:
    def __str__(${1:self})$0:    
    def __reduce__(${1:self})$0:
```

**Comment by ccordoba12** [comment]
@andfoy, please take a look at this one.

**Comment by ccordoba12** [comment]
This was fixed by PR #11400.
